narwhals/_pandas_like/dataframe.py

```diff
@@ -276,16 +276,28 @@ def join(
         if isinstance(right_on, str):
             right_on = [right_on]
 
-        return self._from_dataframe(
-            self._dataframe.merge(
-                other._dataframe,
-                left_on=left_on,
-                right_on=right_on,
-                how=how,
-                suffixes=("", "_right"),
-            ),
+        # performing the merge
+        merged_df = self._dataframe.merge(
+            other._dataframe,
+            left_on=left_on,
+            right_on=right_on,
+            how=how,
+            suffixes=("", "_right"),
         )
 
+        start_col = "a"
+        end_col = "a_right"
```

Comment:
what if someone has a dataframe where the column is called something different than `'a'`?